base/compiler/typeinfer.jl

```diff
@@ -191,6 +191,19 @@ function store_backedges(frame::InferenceState)
                 end
             end
         end
+        edges = frame.src.edges
+        if edges !== nothing
+            Core.println(edges === nothing)
+            Core.println(edges)
+            Core.println(typeof(edges))
+            edges = edges::Vector{MethodInstance}
+            Core.println(edges)
+            for edge in edges
+                @assert isa(edge, MethodInstance)
+                ccall(:jl_method_instance_add_backedge, Cvoid, (Any, Any), edge, caller)
+            end
+            empty!(edges)
```

Comment:
should be non-destructive

Reply:
Do we ever look at this twice? I figure after this the CodeInfo holds the inferred result and we don't ever come back here again (and nobody caches the CodeInfo after it comes out of the generated function).

Reply:
mutating it here though just means we had to make a copy of it earlier though, so it's strictly worse. the memory won't be reclaimed until gc regardless

Reply:
fair. Perhaps what I meant here was `frame.src.edges = nothing`?